g2core/settings/settings_shapeoko2.h

```diff
@@ -156,70 +157,91 @@
 #define Z_LATCH_BACKOFF             4
 #define Z_ZERO_BACKOFF              2
 
-//*** Input / output settings ***
-/*
-    IO_MODE_DISABLED
+/* Legend of valid options:
+
+INPUTS
+---------------------
+  DIn_ENABLED
+    IO_UNAVAILABLE   // input/output is missing/used/unavailable
+    IO_DISABLED      // input/output is disabled
+    IO_ENABLED       // input/output enabled
+
+  DIn_POLARITY:
     IO_ACTIVE_LOW    aka NORMALLY_OPEN
     IO_ACTIVE_HIGH   aka NORMALLY_CLOSED
 
+  DIn_ACTION:
     INPUT_ACTION_NONE
-    INPUT_ACTION_STOP
-    INPUT_ACTION_FAST_STOP
-    INPUT_ACTION_HALT
-    INPUT_ACTION_RESET
-
-    INPUT_FUNCTION_NONE
-    INPUT_FUNCTION_LIMIT
-    INPUT_FUNCTION_INTERLOCK
-    INPUT_FUNCTION_SHUTDOWN
-    INPUT_FUNCTION_PANIC
+    INPUT_ACTION_STOP        - stop at normal jerk - preserves positional accuracy
+    INPUT_ACTION_FAST_STOP   - stop at high jerk - preserves positional accuracy
+    INPUT_ACTION_HALT        - stop immediately - not guaranteed to preserve position
+    INPUT_ACTION_CYCLE_START - start / restart cycle after feedhold (RESERVED)
+    INPUT_ACTION_ALARM       - initiate an alarm. stops everything immediately - preserves position
+    INPUT_ACTION_SHUTDOWN    - initiate a shutdown. stops everything immediately - does not preserve position
+    INPUT_ACTION_PANIC       - initiate a panic. stops everything immediately - does not preserve position
+    INPUT_ACTION_RESET       - reset system
+
+    INPUT_ACTION_LIMIT       - limit switch processing
+    INPUT_ACTION_INTERLOCK   - interlock processing
+
+OUTPUTS
+---------------------
+  DOn_ENABLED
+    IO_UNAVAILABLE   // input/output is missing/used/unavailable
+    IO_DISABLED      // input/output is disabled
+    IO_ENABLED       // input/output enabled
+
+  DOn_POLARITY:
+    IO_ACTIVE_LOW    aka NORMALLY_OPEN
+    IO_ACTIVE_HIGH   aka NORMALLY_CLOSED
+
 */
 // Xmin on v9 board
-#define DI1_MODE                    NORMALLY_CLOSED
+#define DI1_ENABLED                 IO_ENABLED
+#define DI1_POLARITY                NORMALLY_CLOSED
 //#define DI1_ACTION                  INPUT_ACTION_STOP
-#define DI1_ACTION                  INPUT_ACTION_NONE
-#define DI1_FUNCTION                INPUT_FUNCTION_LIMIT
+#define DI1_ACTION                  INPUT_ACTION_LIMIT
 
 // Xmax
```

Comment:
These comments aren’t accurate and can be snipped.
```suggestion
// Xmax
```